main/src/mpu.cpp

```diff
@@ -0,0 +1,157 @@
+#include "mpu.h"
+#include "MPU6050.h"
+
+namespace mpu {
+
+MPU6050 mpudev;
+
+static SemaphoreHandle_t mutex = xSemaphoreCreateMutex();
+static unsigned int count = 0;
+static Data internal{};
+static float gyro_res = 0, accel_res = 0;
+static bool init_success = false;
+
+/**
+ * @brief Mpu initialization. Return gracefully on initialisation error.

```

Comment:
It isn't generated by copilot. I personally use doxygen style comments because vscode supports it for c++. Also the content of comments reflects what we are taught at the uni.